From 61833925b9f8b5ac556f07f28ac17be0ff0aad8e Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 25 Apr 2024 09:15:56 +0200
Subject: [PATCH 0831/1695] core: pta: RTC enumerates upon RTC registration

Allow RTC PTA to be enumerated only upon registration of an
RTC device.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a50fe6b3f349e643f2ed1b3b8717ed701d8e0a1b core: pta: RTC enumerates upon RTC registration]
Change-Id: Ib031021f1f936482be9b37bb417d1eb7ce2ee313
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/377162
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/pta/rtc.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/core/pta/rtc.c b/core/pta/rtc.c
index 8cc6797ab..de2eade56 100644
--- a/core/pta/rtc.c
+++ b/core/pta/rtc.c
@@ -188,8 +188,18 @@ static TEE_Result invoke_command(void *session __unused,
 	return TEE_ERROR_NOT_IMPLEMENTED;
 }
 
+static bool do_enumerate(void)
+{
+#ifdef CFG_DRIVERS_RTC
+	return rtc_device;
+#else
+	return false;
+#endif
+}
+
 pseudo_ta_register(.uuid = PTA_RTC_UUID, .name = PTA_NAME,
 		   .flags = PTA_DEFAULT_FLAGS | TA_FLAG_CONCURRENT |
 			    TA_FLAG_DEVICE_ENUM,
 		   .open_session_entry_point = open_session,
-		   .invoke_command_entry_point = invoke_command);
+		   .invoke_command_entry_point = invoke_command,
+		   .ask_for_enumeration = do_enumerate);
-- 
2.47.2

